src/avatar.jsx

```diff
@@ -156,7 +154,7 @@ const Avatar = React.createClass({
           className={className}
         >
           {icon && React.cloneElement(icon, {
-            color: color,
+            color: color || avatar.color,
```

Comment:
I think the latter is part of a larger discussion - there needs to be a way to style inner components, otherwise users end up having to re-implement components just to add styling - there are already some components where this is an issue, and where applying a local theme may not be sufficient.